src/main/webapp/fundingSourceReport/FundingsSourceReport.js

```diff
@@ -0,0 +1,253 @@
+  }
+
+  init = () => {
+    spinnerService.showAll();
+    this.setState({ isAdmin: component.isAdmin });
+    this.tableHandler(0, this.state.sizePerPage, this.state.search, this.state.sort, this.state.currentPage);
+  };
+
+  tableHandler = (offset, limit, search, sort, page) => {
+    let query = {
+      draw: 1,
+      start: offset,
+      length: limit,
+      orderColumn: sort.orderColumn,
+      sortDirection: sort.sortDirection,
+      searchValue: search,
+    };
+    spinnerService.showAll();
+    Reports.getFundingsReports(query).then(result => {
+      const lastPage = Math.ceil(result.data.recordsFiltered / query.length);
+      this.setState(prev => {
+        prev.lastPage = lastPage;
+        prev.currentPage = page;
+        prev.isAdmin = this.state.isAdmin;
+        prev.fundings = result.data.data;
+        prev.recordsTotal = result.data.recordsTotal;
+        prev.recordsFiltered = result.data.recordsFiltered;
+        prev.sizePerPage = query.length;
+        prev.search = query.searchValue;
+        prev.sort = {
+          orderColumn : query.orderColumn,
+          sortDirection: query.sortDirection
+        };
+        return prev;
+      }, () => spinnerService.hideAll())
+    }).catch(error => {
+      spinnerService.hideAll();
+      this.setState(() => { throw error });
+    });
+  };
+
+  onSearchChange = (search) => {
+    this.tableHandler(0, this.state.sizePerPage, search, this.state.sort, 1);
+  };
+
+  onPageChange = (page, sizePerPage) => {
+    const offset = (page - 1) * sizePerPage;
+    this.tableHandler(offset, sizePerPage, this.state.search, this.state.sort, page);
+  };
+
+  onSizePerPageListHandler = (size) => {
+    this.setState(prev => {
+      prev.query.length = size;
+      return prev;
+    }, () =>{
+      this.tableHandler(this.state.query);
+    });
+  };
+
+  onSortChange = (sortName, sortOrder) => {
+    const sort = {
+      sortDirection: sortOrder,
+      orderColumn: SORT_NAME_INDEX[sortName]
+    };
+    this.tableHandler(0, this.state.sizePerPage, null, sort)
+  };
+
+  onTableChange = (type, newState) => {
+    switch(type) {
+      case TABLE_ACTIONS.SEARCH: {
+        this.onSearchChange(newState.searchText);
+        break
+      }
+      case TABLE_ACTIONS.FILTER: {
+        // Not implemented
+        break;
+      }
+      case TABLE_ACTIONS.PAGINATION: {
+        this.onPageChange(newState.page, newState.sizePerPage);
+        break;
+      }
+      case TABLE_ACTIONS.SORT: {
+        this.onSortChange(newState.sortName, newState.sortOrder);
+        break;
+      }
+    }
+  };
+
+  printContent = () => {
+    let cols = columns.filter(el => el.dataField !== 'id');
+    let fundingsArray = formatData(this.state.fundings, cols);
+    const tableColumnsWidth = [100, 100,'*',80 ,'*','*','*','*','*'];
+    const titleText = "Funding Source Report";
+    printData(fundingsArray, titleText, '', tableColumnsWidth, 'A3', 'landscape');
+  };
+
+  render() {
+    return(
```

Comment:
Thank you for keeping this render very compact. We should strive for this. If renders become huge blobs of html, we need to break those out into smaller components.

Reply:
:bowtie: